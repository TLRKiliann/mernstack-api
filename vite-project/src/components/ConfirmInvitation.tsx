import React from 'react'
import './styleComponents/ConfirmInvitation.scss'

interface ConfirmInvitationProps {
  isChecked: boolean
  handleCheckBox: () => void
  handleValidInvitation: (e: React.MouseEvent<HTMLButtonElement>) => void
}

const ConfirmInvitation: React.FC = (props: ConfirmInvitationProps) => {
  console.log(props, "props")
  return (
    <div className="box--confirminvitation">
      <h4 className="confirm">Confirm invitation ?</h4>
      <label>
        <input
          type="checkbox"
          checked={props.isChecked}
          onChange={props.handleCheckBox}
          className="input--validationInvite"
        />
        yes
      </label>

      <button
        type="button"
        onClick={props.handleValidInvitation}
        className="btn--validationlast"
      >
        Validate
      </button>
    </div>
  )
}

export default ConfirmInvitation