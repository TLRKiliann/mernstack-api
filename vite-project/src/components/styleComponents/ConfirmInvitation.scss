$soft_blue: #93CAED;
$yellow: #ffff00;
$purple: #800080;
$sandybrown: #f4a460;
$orangered: #ff4500;
$whitesmoke: #f5f5f5;
$lightgrey_24: #242424;
$dark_moderate_red: #ae3c52;
$font_dejavu: 'DejaVu Sans', sans-serif;
$font_acme: 'Acme', sans-serif;

.box--confirminvitation {
  position: absolute;
  top: 0px;
  right: 0px;
  left: 0px;
  width: 500px;
  margin: auto;
  margin-top: 23%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
  background: linear-gradient(180deg,
    $sandybrown, $purple);
  border-radius: 15px;
  z-index: 2;
  .confirm {
    margin: auto;
    margin-top: 20px;
    margin-bottom: 20px;
    font-family: $font_dejavu;
    font-size: 1.2rem;
    color: $purple;
    .span--confirmation {
      margin-right: 5px;
      margin-left: 5px;
      font-family: $font_acme;
      font-size: 1.8rem;
      text-shadow: 2px 1px 1px $purple;
      color: $soft_blue;
    }
  }
  .checkbox--div {
    width: 50%;
    margin: auto;
    display: flex;
    flex-direction: row;
    justify-content: space-around;
  }
  .input--validationInvite {
    background-color: $lightgrey_24;
    color: $whitesmoke;
    border: 1px solid $yellow;
  }
  .btn--validationlast {
    margin-top: 20px;
    margin-bottom: 20px;
    padding: 5px 10px;
    font-family: $font_dejavu;
    font-size: 0.9rem;
    font-weight: bold;
    background: linear-gradient(180deg,
      $sandybrown, $orangered, $dark_moderate_red);
    border: 1px solid $purple;
    border-radius: 7px;
    color: $whitesmoke;
    &:hover {
      background: $sandybrown;
      color: $dark_moderate_red;
    }
  }
}