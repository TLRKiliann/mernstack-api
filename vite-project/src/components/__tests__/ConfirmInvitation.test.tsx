import React from "react";
import { screen, render } from '@testing-library/react';
import { test, expect, vi } from 'vitest'
import { create } from 'react-test-renderer'
import '@testing-library/jest-dom'
import ConfirmInvitation from "../ConfirmInvitation.tsx";



test('my test for custom-element id', () => {
  render(<ConfirmInvitation />);
  const confirmtestid = screen.getByTestId("confirmtestid");
  expect(confirmtestid).toBeInTheDocument();
})

test('MatchSnapShot test ConfirmInvitation', () => {
  const treeConfirmInvitation = create(<ConfirmInvitation />)
  expect(treeConfirmInvitation.toJSON()).toMatchSnapshot()
})