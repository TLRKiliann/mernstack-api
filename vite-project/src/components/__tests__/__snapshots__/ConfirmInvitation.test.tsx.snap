// Vitest Snapshot v1

exports[`MatchSnapShot test ConfirmInvitation 1`] = `
<div
  className="box--confirminvitation"
  data-testid="confirmtestid"
>
  <h4
    className="confirm"
  >
    <span
      className="span--confirmation"
    >
       
    </span>
    do you confirm invitation ?
  </h4>
  <div
    className="checkbox--div"
  >
    <label>
      <input
        className="input--validationInvite"
        type="checkbox"
      />
       Yes
    </label>
    <label>
      <input
        className="input--validationInvite"
        type="checkbox"
      />
       No
    </label>
  </div>
  <button
    className="btn--validationlast"
    data-testid="btnconfirminvitation"
    type="button"
  >
    Validate
  </button>
</div>
`;
